### DISCUSSION TITLE
/blog/2021/12/01/multilevel-models-panel-data-guide/

### DISCUSSION THREAD

**Discussion by andrewheiss** [andrewheiss/ath-quarto#49] [Blog comments]
A guide to working with country-year panel data and Bayesian multilevel models | Andrew Heiss 
 https://www.andrewheiss.com/blog/2021/12/01/multilevel-models-panel-data-guide/

**Comment by andrewheiss** [comment]
```header
Originally posted by Tiny van Boekel on 2022-09-05 19:21:01
````
Great post, as are all the others! very clear and instructive. However,
could it be that there is a small mistake where you calculate
level-variance as a percentage of total variance (basically ICC,
intra-class correlation)? It seems that you took the brms output for the
calculation, which gives standard errors and not variances; so the
standard errors should be squared in my humble opinion to do this
calculation. Also, I wonder whether you know the R package performance
that does this type of calculations specifically for brms-fitted models
with variance decomposition, that is supposed to be equivalent to ICC
see(
[https://easystats.github.io\...](https://easystats.github.io/performance/reference/icc.html?q=variance_de#variance-decomposition-for-brms-models) &quot;https://easystats.github.io/performance/reference/icc.html?q=variance_de#variance-decomposition-for-brms-models)&quot;)

**Reply by andrewheiss** [reply]
```header
Originally posted on 2022-09-15 19:55:55
````
Oh neat! I hadn\'t seen the performance package yet---that looks really
helpful!

**Comment by andrewheiss** [comment]
```header
Originally posted by Corey Neskey on 2022-11-24 22:02:16
````
Thank you for the great post!

`element_text(family = &quot;Consolas&quot;)`\
I didn\'t have this font and had to dig into error messages to find out
why the code wouldn\'t run. You may want to swap it for something
standard for easier following.

For anyone who gets errors installing rstan on a Mac with M1 be sure to
download xcode from the appstore. Doing so with brew or directly from
terminal didnt work for some reason

**Comment by andrewheiss** [comment]
```header
Originally posted by Gianni Lombardo on 2022-11-28 07:53:19
````
Thanks for posting this.\
Two questions (I\'m an economist, ie coming from the econometric
perspective, notation etc)\
1. the notation b\_{0,country} and b\_{1,country} is confusing me. They
are not interacted with anything and are country specific. How can they
be identified? Is it just notation?\
2. I\'ve been held back from using Bayesian methods for panels as I\'m
not sure how to control for eg clustering, heteroskedasticity etc
(robust se) in this context. Frequentist approaches offer a number of
ways to correct for these.

Any suggestion on where to read about the latter issue?\
Thanks

**Reply by andrewheiss** [reply]
```header
Originally posted on 2022-11-29 05:25:13
````
For (1), check out chapters 15--17 of *Bayes Rules*
([https://www.bayesrulesbook\....](https://www.bayesrulesbook.com/chapter-15.html) &quot;https://www.bayesrulesbook.com/chapter-15.html)&quot;),
which has the best overview of multilevel models I\'ve found (even if
you don\'t use Bayesian methods). They cover the b_0 offset notation
(see 16.3.2:
[https://www.bayesrulesbook\....](https://www.bayesrulesbook.com/chapter-16.html#another-way-to-think-about-it) &quot;https://www.bayesrulesbook.com/chapter-16.html#another-way-to-think-about-it)&quot;)
as well as an alternative type of notation for random effects

For (2), multilevel models in general (both frequentist and Bayesian)
don\'t really use robust se (see
[https://statmodeling.stat.c\...](https://statmodeling.stat.columbia.edu/2007/11/28/clustered_stand/) &quot;https://statmodeling.stat.columbia.edu/2007/11/28/clustered_stand/)&quot;).
Rather than account for the clustered structure of the data through the
standard errors, the multilevel approach models the clustering directly

**Comment by janpfander** [comment]
This is amazing, l check this regularly to remind myself of what's the crack with random effects, thank you! 
A minor comment: When you write the formula, shouldn't it say: 
(\beta_1 + b_{1, Country}) \text{Year} instead of (\beta_1  \text{Year}  + b_{1, Country}) ?

**Reply by andrewheiss** [reply]
Yes, good catch! I wrote this post before I knew a lot about multilevel models, so I got that part wrong! I've updated it here in the post. 

You also might want to check out [chapters 15–18 in *Bayes Rules!*](https://www.bayesrulesbook.com/chapter-17.html), which I think is the absolute best explanation of multilevel models I've ever found. (And I have [a brms translation of it here](https://bayesf22-notebook.classes.andrewheiss.com/bayes-rules/17-chapter.html) where I use the correct notation 😂)

**Reply by janpfander** [reply]
Thanks for the references, that looks great!

**Comment by cjsewall9** [comment]
Fantastic post, Andrew! Thanks for all your hard work. Since you’re coming from an Econ background, I’m wondering if you can help me—someone coming from a psych background—translate the “two-way fixed effects” model Econ model in multilevel modeling terms. I believe you touch on this in your post but don’t make a direct/explicit comparison (unless I missed it). Is it fair to say that a TWFE model is (nearly) equivalent to a multilevel model with random intercepts for unit and time?

**Reply by andrewheiss** [reply]
Ha I wish. I have another post about econ-style fixed effects and how they're weird: https://www.andrewheiss.com/blog/2022/11/29/conditional-marginal-marginaleffects/

(and a post about TWFE specifically: https://www.andrewheiss.com/blog/2021/08/25/twfe-diagnostics/ )

The general idea of accounting for unit and time is the same, but the approaches are different (and the econ world tends to hate the idea of random effects). Roughly speaking, TWFE is just `y ~ x + country + as.factor(year)`

**Comment by DjakissO** [comment]
Hi Professor,

Thank you for this helpful post.
I want to fit a Dynamic Panel Ordered Probit Model. But I do not find a package adapted for it. I want to know if your code can support ordered probit, and dynamic probit.
